File: app/src/main/java/ao/co/isptec/aplm/sca/dto/NewIncidentRequest.java
package ao.co.isptec.aplm.sca.dto;

import com.google.gson.Gson;

public class NewIncidentRequest {
    private String title;
    private String description;
    private Double latitude;
    private Double longitude;
    private String datetime; // ISO-8601 string
    private Boolean urgency;  // boolean as backend expects
    private Long userId;

    public NewIncidentRequest(String title, String description, Double latitude, Double longitude,
                              String datetime, Boolean urgency, Long userId) {
        this.title = title;
        this.description = description;
        this.latitude = latitude;
        this.longitude = longitude;
        this.datetime = datetime;
        this.urgency = urgency;
        this.userId = userId;
    }

    public String toJson() {
        return new Gson().toJson(this);
    }

    // Getters
    public String getTitle() { return title; }
    public String getDescription() { return description; }
    public Double getLatitude() { return latitude; }
    public Double getLongitude() { return longitude; }
    public String getDatetime() { return datetime; }
    public Boolean getUrgency() { return urgency; }
    public Long getUserId() { return userId; }
}

File: api/src/main/kotlin/ao/co/isptec/aplm/cidadaoactivo/api/dtos/NewIncidentRequest.kt
package ao.co.isptec.aplm.cidadaoactivo.api.dtos

import org.springframework.web.multipart.MultipartFile
import java.time.LocalDateTime

data class NewIncidentRequest(
    val title: String,
    val description: String,
    val latitude: Double,
    val longitude: Double,
    val urgency: Boolean,
    val datetime: String,
    val userId: Long
)
